### PR TITLE
Target Noise Regularization: Gaussian noise on training targets

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Adding small Gaussian noise to training targets prevents the model from memorizing exact training values, forcing it to learn smoother, more generalizable representations. This is a classic regularization technique (label smoothing for regression) that is especially effective when the model is close to overfitting the training distribution.

**Why now:** Our baseline has been heavily tuned over 136 merged PRs. The model likely fits the training data very well — but our OOD metrics (p_oodc 7.64 vs ensemble 6.6, p_re 6.42 vs ensemble 5.8) suggest overfitting to the training distribution. Adding target noise acts as implicit Bayesian regularization — it's equivalent to training with a data augmentation that smears the target distribution, encouraging the model to predict the conditional mean rather than memorizing individual samples.

**Evidence:** Label smoothing / target noise is one of the most reliable regularization techniques in Kaggle competitions for structured regression. Szegedy et al. (CVPR 2016) showed label smoothing improved ImageNet accuracy; the regression analog (Gaussian target noise) has similar benefits for OOD robustness.

**Key insight:** The noise should be proportional to local target variance — high-pressure regions can tolerate more noise than low-pressure regions. We'll use relative noise: noise_std = sigma * |target_value|.

### Step 1: Add arguments

```python
parser.add_argument('--target_noise', type=float, default=0.0,
                    help='Std of Gaussian noise added to targets during training (relative to |target|)')
```

### Step 2: Apply noise during training only

After loading targets but before loss computation:

```python
if args.target_noise > 0 and model.training:
    # Relative Gaussian noise: proportional to |target|
    noise_std = args.target_noise * y.abs().clamp(min=0.01)  # Floor to prevent zero noise
    noise = torch.randn_like(y) * noise_std
    y_noisy = y + noise
    # Use y_noisy for loss computation
    y = y_noisy
```

**Important:** Do NOT apply noise during validation — metrics must be computed on clean targets.

### Step 3: Run 2 seeds with sigma=0.01

Start with small noise (1% relative):

```bash
cd cfd_tandemfoil && python train.py \
  --agent tanjiro --wandb_name "tanjiro/target-noise-0.01-s42" --seed 42 \
  --wandb_group target-noise-regularization \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 150 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature --re_stratified_sampling \
  --target_noise 0.01

# Repeat for seed 73: --seed 73 --wandb_name "tanjiro/target-noise-0.01-s73"
```

**Compute overhead:** Near zero — one randn_like per batch.

## Baseline

| Metric | Baseline | Target to beat |
|--------|----------|----------------|
| p_in | 11.742 | < 11.742 |
| p_oodc | 7.643 | < 7.643 |
| p_tan | 27.874 | < 27.874 |
| p_re | 6.419 | < 6.419 |

- **Baseline W&B runs:** k5qwvce4 (seed 42), 7oa5xfhi (seed 73)
- **val/loss baseline:** ~0.37

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT tanjiro:

## Results

### Target Noise Regularization (sigma=0.01, relative Gaussian noise on raw targets)

| Metric | Baseline | Seed 42 | Seed 73 | 2-seed avg | Δ |
|--------|----------|---------|---------|------------|---|
| **p_in** | 11.742 | 11.476 | 11.867 | **11.672** | **-0.6%** ✅ |
| **p_oodc** | 7.643 | 7.545 | 7.393 | **7.469** | **-2.3%** ✅ |
| **p_tan** | 27.874 | 28.675 | 28.340 | **28.508** | **+2.3%** ❌ |
| **p_re** | 6.419 | 6.367 | 6.244 | **6.306** | **-1.8%** ✅ |
| val_loss | ~0.37 | 0.3792 | 0.3742 | 0.3767 | +1.8% |

### Surface MAE (full breakdown, 2-seed avg)

| Split | Ux | Uy | p |
|-------|----|----|---|
| val_in_dist | 2.497 | 0.793 | 11.672 |
| val_ood_cond | 1.976 | 0.595 | 7.469 |
| val_tandem_transfer | 4.042 | 1.542 | 28.508 |
| val_ood_re | 1.697 | 0.416 | 6.306 |

### Volume MAE (2-seed avg)

| Split | Ux | Uy | p |
|-------|----|----|---|
| val_in_dist | 0.517 | 0.191 | 12.084 |
| val_ood_cond | 0.343 | 0.146 | 6.544 |
| val_tandem_transfer | 1.297 | 0.604 | 31.604 |
| val_ood_re | 0.503 | 0.269 | 5.631 |

### Commands used

```bash
# Seed 42
cd cfd_tandemfoil && CUDA_VISIBLE_DEVICES=0 python train.py \
  --agent tanjiro --wandb_name "tanjiro/target-noise-0.01-s42" --seed 42 \
  --wandb_group target-noise-regularization \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 150 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature --re_stratified_sampling \
  --target_noise 0.01

# Seed 73 — same but --seed 73 --wandb_name "tanjiro/target-noise-0.01-s73"
```

### W&B runs
- Seed 42: `a68uozh3` — https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/a68uozh3
- Seed 73: `ik8t4w8i` — https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/ik8t4w8i

### Analysis

**Mixed results.** The relative target noise regularization improves 3 of 4 OOD/generalization metrics but regresses tandem transfer:

- **p_oodc -2.3%** and **p_re -1.8%**: The hypothesis that target noise acts as implicit Bayesian regularization for OOD robustness is partially confirmed. Both OOD metrics improved meaningfully.
- **p_in -0.6%**: Slight in-distribution improvement, consistent with regularization benefit.
- **p_tan +2.3%**: The tandem transfer split regressed. This makes sense — tandem samples have more complex pressure distributions (multi-foil interactions), and adding noise proportional to |target| injects disproportionately more noise into the high-pressure regions of tandem cases. The `clamp(min=0.01)` floor means even low-pressure regions get non-trivial noise.

The val_loss also regressed slightly (+1.8%), which is expected — the model is being trained on noisier targets so it can't fit the clean validation as tightly.

**Note:** This relative noise (`--target_noise`) operates on raw targets *before* normalization, while the existing fixed-schedule noise operates on normalized targets. Both were active in this experiment. The interaction may be suboptimal.

### Suggested follow-ups

1. **Tandem-aware noise scaling:** Apply reduced noise (or zero noise) to tandem samples to preserve p_tan while keeping the OOD benefit. The `is_tandem` flag is already computed in the training loop.
2. **Try sigma=0.005:** Half the noise level might preserve the OOD gains while reducing the p_tan regression.
3. **Disable existing fixed noise when using target_noise:** Running `--no_target_noise --target_noise 0.01` to isolate the relative noise effect and avoid double-noising.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2332 because ADVISOR: Closing — Target noise regularization shows mixed results (p_oodc -2.3%, p_re -1.8% but p_tan +2.3%). Additionally, this experiment was run against a different base configuration (includes tandem_ramp, domain_layernorm, aug_gap_stagger_sigma, etc.) rather than the current baseline, making direct comparison unreliable. The p_tan regression is concerning since tandem transfer is our weakest metric. Target noise was already classified as neutral in prior rounds. Good analysis of the tandem-aware noise idea, but the fundamental challenge (noise proportional to |target| disrupts high-pressure tandem regions) is not easily resolvable. This direction is exhausted..
